Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -107,6 +107,7 @@ void AP_Periph_FW::init()
 
 #ifdef HAL_PERIPH_ENABLE_GPS
     if (gps.get_type(0) != AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+        serial_manager.set_protocol_and_baud(g.gps_port, AP_SerialManager::SerialProtocol_GPS, AP_SERIALMANAGER_GPS_BAUD);
```

Comment:
we'll also need to ensure any ports below that number are not GPS, as AP_GPS looks for first port that is a GPS. Or we could change AP_SerialManager to default protocol to zero for AP_Periph

Reply:
Maybe the serial manager init needs to change for periph build so it sets all ports as None instead of the standard vehicle configs of telem 1, telem 2, gps1...

Reply:
Actually.. all we have to do is NOT call serial_manager.init() in AP_Periph_FW::init(). I'm not sure what purpose it serves after this PR goes in anyways... Although we'll need to call set_protocol_and_baud for every uart we begin().. like ADSB, HWESC, MSP, serial 0 and 1. Hmm, maybe there's another way.

Reply:
@tridge I added a crude brute-force solution. Please take a look

Reply:
Reworked using defines